### PR TITLE
Amls 5540

### PR DESCRIPTION
**Pull request by ajchilts**
Wire up AMLS FE and EAB FE 

Acceptance criteria: Full round trip minus DES (API) submissions. 

## Related / Dependant PRs?

https://github.com/hmrc/amls-estate-agency-business-frontend/pull/21

## How Has This Been Tested?

Local both toggled on and off. Unit tests + Regression with toggle off. 

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [ ] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [ ] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: test/controllers/eab/EabControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+      val result = controller.get(credId)(request)
+      status(result) must be(OK)
+
+      val document = Json.parse(contentAsString(result))
+      document mustBe(Json.obj())
+    }
+
+    "amp section in cache" in new Fixture {
+      when(proxyCacheService.getEab(any())(any())).thenReturn(Future.successful(Some(completeEabJson)))
```
Comment (slavvco) [inline test/controllers/eab/EabControllerSpec.scala]: amp -> eab

File: app/controllers/amp/AmpController.scala [hunk 2]
```diff
@@ -24,27 +24,27 @@ import javax.inject.Inject
 import models.amp.Amp
 import play.api.libs.json._
 import play.api.mvc.{Action, MessagesControllerComponents}
-import services.amp.AmpCacheService
+import services.amp.ProxyCacheService
 import utils.AuthAction
```
Comment (slavvco) [inline app/controllers/amp/AmpController.scala]: empty line above

File: app/controllers/eab/EabController.scala [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import controllers.{AmlsBaseController, CommonPlayDependencies}
+import javax.inject.Inject
+import models.eab.Eab
+import play.api.libs.json._
+import play.api.mvc.MessagesControllerComponents
+import services.amp.ProxyCacheService
+import utils.AuthAction
+
+
+class EabController @Inject()(proxyCacheService  : ProxyCacheService,
```
Comment (slavvco) [inline app/controllers/eab/EabController.scala]: empty line above

File: app/models/eab/Eab.scala [hunk 4]
```diff
@@ -76,18 +77,28 @@ final case class Eab(data: JsObject = Json.obj(),
...
+
+  def isInvalidRedressScheme: Boolean = {
+    val scheme = get[String](Eab.redressScheme)
+    scheme match {
+      case Some(x) if x.nonEmpty && x.contains("other") => true
+      case Some(x) if x.nonEmpty && x.contains("ombudsmanServices") => true
+      case _ => false
+    }
+  }
+
```
Comment (slavvco) [inline app/models/eab/Eab.scala]: unnecessary empty line

File: app/utils/ControllerHelper.scala [hunk 5]
```diff
@@ -43,13 +44,21 @@ object ControllerHelper {
...
+  //TODO - can be removed when we remove the old EAB models
+  def hasInvalidRedressScheme(eabOpt: Option[EstateAgentBusiness]): Boolean = {
     eabOpt.exists(eab =>
       eab.redressScheme.exists(redressScheme =>
         redressScheme match {
           case OmbudsmanServices | Other(_) => true
           case _  => false
         }))
+  }
+  def hasInvalidRedressSchemeNewEab(eabOpt: Option[Eab]): Boolean = {
```
Comment (slavvco) [inline app/utils/ControllerHelper.scala]: please separate  method definitions adding blank line between 55-56

File: app/controllers/amp/AmpController.scala (hunk 2 above)
Comment (ajchilts) [reply app/controllers/amp/AmpController.scala]: Harsh :-P

File: app/models/eab/Eab.scala (hunk 4 above)
Comment (ajchilts) [reply app/models/eab/Eab.scala]: Really harsh :-P

File: app/utils/ControllerHelper.scala (hunk 5 above)
Comment (ajchilts) [reply app/utils/ControllerHelper.scala]: When overriding a method that will soon be removed I always leave no gap. But I can add a gap if it makes you happy. 👍

File: test/controllers/eab/EabControllerSpec.scala [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+      .overrides(bind[AuthAction].to(SuccessfulAuthAction))
+      .overrides(bind[ProxyCacheService].to(proxyCacheService))
+      .overrides(bind[DataCacheConnector].to(mockCacheConnector))
+      .build()
+
+    val controller      = app.injector.instanceOf[EabController]
+  }
+
+  "get returns 200" when {
+    "no amp section in cache" in new Fixture {
```
[the remark this answers is not in the document]
Comment (ajchilts) [reply test/controllers/eab/EabControllerSpec.scala]: Good spot (captain cut and paste)

File: test/controllers/eab/EabControllerSpec.scala (hunk 1 above)
Comment (ajchilts) [reply test/controllers/eab/EabControllerSpec.scala]: Good spot (captain cut and paste)

File: app/controllers/amp/AmpController.scala (hunk 2 above)
Comment (slavvco) [reply app/controllers/amp/AmpController.scala]: haha, sorry :)

File: app/utils/ControllerHelper.scala (hunk 5 above)
Comment (slavvco) [reply app/utils/ControllerHelper.scala]: right if that's the case let's leave it

**Pull request merged**